### PR TITLE
generator: remove `CHECK` abort calls exposed by public API

### LINKED ISSUE #49: Remove `CHECK` on blinding factor from generator/main_impl.h
See https://github.com/ElementsProject/elements/issues/540

### PR DESCRIPTION
**Pull request by apoelstra**
Fixes #49

### CODE REVIEW

**Comment by jonasnick** [comment]
ACK d5839786b325e7ba0f89a7a2c9809971890a4ec6

File: src/modules/generator/tests_impl.h [hunk 1]
```diff
@@ -188,6 +188,14 @@ void test_generator_generate(void) {
         secp256k1_ge_to_storage(&ges, &ge);
         CHECK(memcmp(&ges, &results[i - 1], sizeof(secp256k1_ge_storage)) == 0);
     }
+
+    /* There is no range restriction on the value, but the blinder must be a
+     * valid scalar. Check that an invalid blinder causse the call to fail,
```
Comment (instagibbs) [inline src/modules/generator/tests_impl.h]: s/causse/cause/

s/but not a/but not/

Comment (jonasnick) [reply src/modules/generator/tests_impl.h]: ping @apoelstra

Comment (apoelstra) [reply src/modules/generator/tests_impl.h]: fixed

**Comment by jonasnick** [comment]
reACK 2def4ace2ca074ab781b9739c395c5997c843bbb

**Comment by instagibbs** [comment]
post-hom ACK

**Pull request merged**